modules/contact/src/userobjects/LMWeightedGapUserObject.C

```diff
@@ -0,0 +1,48 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LMWeightedGapUserObject.h"
+#include "MooseVariableFE.h"
+#include "SystemBase.h"
+
+registerMooseObject("ContactApp", LMWeightedGapUserObject);
+
+InputParameters
+LMWeightedGapUserObject::validParams()
+{
+  InputParameters params = WeightedGapUserObject::validParams();
+  params.addRequiredCoupledVar(
+      "lm_variable", "The Lagrange multiplier variable representing the contact pressure.");
+  return params;
+}
+
+LMWeightedGapUserObject::LMWeightedGapUserObject(const InputParameters & parameters)
+  : WeightedGapUserObject(parameters), _lm_var(getVar("lm_variable", 0))
+{
+  if (!_lm_var)
+    paramError("lm_variable",
+               "The Lagrange multiplier variable must be an actual variable and not a constant.");
```

Comment:
`_lm_var` being `nullptr` could also mean that the variable does not exist! (e.g. the user made a typo). We have `isCoupledConstant` now (added in #22598) to disambiguate those cases ... if you want to improve the error in the future.